### PR TITLE
Add command-line options to control SSL certificate verification

### PR DESCRIPTION
**Pull request by ahills**
Adds two options to both `pdc` and `pdc_client`:

-k/--insecure: disable verification (`ssl_verify = False`)
--ca-cert: path to CA certificate (`ssl_verify = '/path/to/CA.crt'`)

When "insecure" behavior is desired, urllib warnings are muted.

### CODE REVIEW

**Comment by ahills** [comment]
@lubomir, @ycheng-aa Would you mind reviewing this patch for me?

File: pdc_client/runner.py [hunk 1]
```diff
@@ -143,7 +146,13 @@ def setup(self):
 
     def run(self, args=None):
         self.args = self.parser.parse_args(args=args)
-        self.client = pdc_client.PDCClient(self.args.server, page_size=self.args.page_size)
+        ssl_verify = self.args.ca_cert if self.args.ca_cert else not self.args.insecure
+        if self.args.insecure:
+            import requests
```
Comment (ycheng-aa) [inline pdc_client/runner.py]: Same as above import.

File: pdc_client/runner.py [hunk 2]
```diff
@@ -125,6 +125,9 @@ def setup(self):
         self.parser = argparse.ArgumentParser(description='PDC Client')
         self.parser.add_argument('-s', '--server', default='stage',
                                  help='API URL or shortcut from config file')
+        self.parser.add_argument('-k', '--insecure', action='store_true',
+                                 help='Disable SSL certificate verification')
+        self.parser.add_argument('--ca-cert', help='Path to CA certificate')
```
Comment (ycheng-aa) [inline pdc_client/runner.py]: Same as above.

File: bin/pdc_client [hunk 3]
```diff
@@ -163,7 +166,12 @@ if __name__ == "__main__":
                 data[key] = ''
 
     try:
-        client = PDCClient(options.server)
+        ssl_verify = args.ca_cert if args.ca_cert else not args.insecure
+        if args.insecure:
+            import requests
```
Comment (ycheng-aa) [inline bin/pdc_client]: Is this imported in top of file a little better?

File: bin/pdc_client [hunk 4]
```diff
@@ -119,6 +119,9 @@ def load_data(options):
 if __name__ == "__main__":
     parser = optparse.OptionParser(version=__version__)
     parser.add_option("-s", "--server", help="PDC instance url or shortcut.")
+    parser.add_option("-k", "--insecure", action="store_true",
+                      help="Disable SSL certificate verification")
+    parser.add_option("--ca-cert", help="Path to CA certificate")
```
Comment (ycheng-aa) [inline bin/pdc_client]: This is path to a CA certificate file or CA certificate folder or both? Could add a little more text to describe it?

File: bin/pdc_client [hunk 5]
```diff
@@ -163,7 +166,12 @@ if __name__ == "__main__":
                 data[key] = ''
 
     try:
-        client = PDCClient(options.server)
+        ssl_verify = args.ca_cert if args.ca_cert else not args.insecure
```
Comment (simozhan) [inline bin/pdc_client]: ca_cert or insecure should be in options, not args which is different with argparse

**Comment by simozhan** [comment]
BTW, it seems when i running command 'bin/pdc -s dt --insecure release list', the output is Warning: 'insecure' option is deprecated; please use 'ssl-verify' insteadRelease-ID                Name                                Activity
The warning is concated with the plugins' output. This is not related to this fix, but i am i guessing we should add print() before every plugins's output in another patch.

**Comment by ycheng-aa** [comment]
Another thing is how to let QA know this new feature should be tested. @ahills  should a JIRA issue be created in our JIRA board ? If you add QE_NEEDED label, QA will test it. Or can you suggest other workflow? I think QA keep testing new feature or bug fix is more reiiable.

**Comment by bliuredhat** [comment]
@ahills ; please help to update this patch, thank you :).

**Comment by ahills** [comment]
Sorry for the delay. I rebased and committed changes in response to comments.

**Comment by ycheng-aa** [comment]
Looks good.

It's better to squash to one commit to merge.

**Comment by lubomir** [comment]
Is including the `.egg-info` directory intentional?

File: pdc_client.egg-info/PKG-INFO [hunk 6]
```diff
@@ -0,0 +1,16 @@
+Metadata-Version: 1.1
```
Comment (bliuredhat) [inline pdc_client.egg-info/PKG-INFO]: @ahills; the `pdc_client.egg-info` folder and the files which under this folder, I consider they are not necessary in this patch. Do you have other purpose to add this folder?

**Comment by ahills** [comment]
@lubomir, great catch. @bliuredhat, I removed it in 9507bc5.

**Comment by ahills** [comment]
This is superceded by #88 (and possibly other changes).

**Pull request closed without merging**